src/tactic/interactive.lean

```diff
@@ -1237,10 +1237,10 @@ propagate_tags $
 do let (p, x) := p,
    e ← i_to_expr p,
    some h ← pure h | tactic.generalize' e x >> skip,
+   -- `h` is given, the regular implementation of `generalize` works.
```

Comment:
I sometimes get lost between `generalize`, `generalize'`, `generalizes`, `generalize_hyp`. Is the idea that `generalize` is okay to use here because the code a few lines below handles the assertion that `generalize` would do?

Reply:
That is fair. I'm happy to expand docstrings if that would help.

`tactic.generalize` fails if it cannot find the generalized expression in the target. However, the case-split below takes care of that case.

Reply:
I should probably just read the existing ones more carefully...